docs/excel/excel-add-ins-workbooks.md

```diff
@@ -0,0 +1,168 @@
+
+```js
+Excel.run(function (context) {
+	var customDocProperties = context.workbook.properties.custom;
+	customDocProperties.add("Introduction", "Hello");
+	return context.sync();
+}).catch(errorHandlerFunction);
+
+[...]
+
+Excel.run(function (context) {
+	var customDocProperties = context.workbook.properties.custom;
+	var customProperty = customDocProperties.getItem("Introduction");
+	customProperty.load("key, value");
+
+	return context.sync().then(function() {
+        console.log("Custom key  : " + customProperty.key); // "Introduction"
+        console.log("Custom value : " + customProperty.value); // "Hello"
+	});
+}).catch(errorHandlerFunction);
+```
+
+## Document settings
+
+A workbook's settings are similar to the collection of custom properties. The different is settings are unique to a single Excel file and add-in pairing, whereas properties are solely connected to the file. The following example shows how to create and access a setting.
+
+```js
+Excel.run(function (context) {
+	var settings = context.workbook.settings;
+	settings.add("NeedsReview", true);
+	var needsReview = settings.getItem("NeedsReview");
+	needsReview.load("value");
+
+	return context.sync().then(function() {
+		console.log("Workbook needs review : " + needsReview.value);
+	});
+}).catch(errorHandlerFunction);
+```
+
+## Calculations
+
+By default, Excel recalculates formula results whenever a referenced cell is changed. Your add-in's performance may benefit from adjusting this calculation behavior. The application object has a `calculationMode` property of type `CalculationMode`. It can be set to the following values:
+
+ - `automatic`: The default recalculation behavior where Excel calculates new formula results every time the relevant data is changed.
+ - `automaticExceptTables`: Same as `automatic`, except any changes made to values in tables are ignored.
+ - `manual`: Calculations only occur when the user or add-in requests them.
+ 
+Application objects provide a method to force an immediate recalculation. `Application.calculate(calculationType)` starts a manual recalculation based on the provided `CalculationType`. The following values can be provided:
+
+ - `full`: Recalculate all formulas in all open workbooks, regardless of whether they have changed since the last recalculation.
+ - `fullRebuild`: Check dependent formulas, and then recalculate all formulas in all open workbooks, regardless of whether they have changed since the last recalculation.
+ - `recalculate`: Recalculate formulas that have changed (or been programmatically marked for recalculation) since the last calculation, and formulas dependent on them, in all active workbooks.
+ 
+The article [Change formula recalculation, iteration, or precision](https://support.office.com/article/change-formula-recalculation-iteration-or-precision-73fc7dac-91cf-4d36-86e8-67124f6bcce4) provides more context on the recalculation terminology.
```

Comment:
Referring to this end-user article is a good idea. Can we do the same thing above in the Protection section?